src/plugins/console/public/lib/osd/api.ts

```diff
@@ -29,12 +29,14 @@
  */
 
 import _ from 'lodash';
-import { UrlPatternMatcher } from '../autocomplete/components';
+import { SharedComponent, UrlPatternMatcher } from '../autocomplete/components';
 import { UrlParams } from '../autocomplete/url_params';
 import {
   globalsOnlyAutocompleteComponents,
   compileBodyDescription,
 } from '../autocomplete/body_completer';
+import { Endpoint } from '../autocomplete/types';
+import { ParametrizedComponentFactories } from './osd';
```

Comment:
if a type file is defined, should move `ParametrizedComponentFactories` in a single type file

Reply:
Will do